compiler/rustc_const_eval/src/transform/check_consts/check.rs

```diff
@@ -384,11 +384,11 @@ impl Checker<'mir, 'tcx> {
                         match pred.skip_binder() {
                             ty::ExistentialPredicate::AutoTrait(_)
                             | ty::ExistentialPredicate::Projection(_) => {
-                                self.check_op(ops::ty::TraitBound(kind))
+                                self.check_op(ops::ty::DynTrait(kind))
                             }
                             ty::ExistentialPredicate::Trait(trait_ref) => {
                                 if Some(trait_ref.def_id) != self.tcx.lang_items().sized_trait() {
```

Comment:
I just noticed: isn't this check redundant? `Sized` trait can't be made into an object anyway.

Reply:
They can't be made into an object (yet, at least), but user's can write all kinds of broken code. Checks like these are "defense in depth" by silencing errors that would otherwise be redundant. Now, in those cases it is good style to leave a comment stating this, and I can't know _for sure_ that that's what happened here and @ecstatic-morse['s code is a year old](https://github.com/rust-lang/rust/commit/3569bb63231979c89b71736cd8b9399aad249784#diff-430f3a60a526044dff4ee953a9f6f25691db98573ecd05c67642488f24a0c153R312) so it's unreasonable to expect them to remember 😊

Let's keep it as is.

Reply:
Ah, so this protects users from seeing both the "`Sized` cannot be made into an object" _and_  "trait objects in const fn are unstable" at the same time? That's nice! I thought that was just copy-pasted from the [bounds check](https://github.com/rust-lang/rust/commit/3569bb63231979c89b71736cd8b9399aad249784#diff-430f3a60a526044dff4ee953a9f6f25691db98573ecd05c67642488f24a0c153R349) 😄

Reply:
That might *also* be the case. Regardless of what the _intent_ was, that's the effect, and I want us to keep it 😅